extensions/pdh/PerformanceDataMonitor.cpp

```diff
@@ -249,32 +265,56 @@ void PerformanceDataMonitor::addCustomPDHCountersFromProperty(const std::string&
   }
 }
 
-void PerformanceDataMonitor::setupMembersFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
-  std::string predefined_groups;
-  if (context->getProperty(PredefinedGroups.getName(), predefined_groups)) {
-    logger_->log_trace("Predefined group configured to be %s", predefined_groups);
-    addCountersFromPredefinedGroupsProperty(predefined_groups);
-  }
-
+void PerformanceDataMonitor::setupCountersFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
   std::string custom_pdh_counters;
   if (context->getProperty(CustomPDHCounters.getName(), custom_pdh_counters)) {
     logger_->log_trace("Custom PDH counters configured to be %s", custom_pdh_counters);
     addCustomPDHCountersFromProperty(custom_pdh_counters);
   }
+}
 
+void PerformanceDataMonitor::setupPredefinedGroupsFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
+  std::string predefined_groups;
+  if (context->getProperty(PredefinedGroups.getName(), predefined_groups)) {
+    logger_->log_trace("Predefined group configured to be %s", predefined_groups);
+    addCountersFromPredefinedGroupsProperty(predefined_groups);
+  }
+}
+
+void PerformanceDataMonitor::setupOutputFormatFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
   std::string output_format_string;
   if (context->getProperty(OutputFormatProperty.getName(), output_format_string)) {
-    if (output_format_string == OPEN_TELEMETRY_FORMAT_STR) {
-      logger_->log_trace("OutputFormat is configured to be OpenTelemetry");
+    if (output_format_string == PRETTY_OPEN_TELEMETRY_FORMAT_STR || output_format_string == COMPACT_OPEN_TELEMETRY_FORMAT_STR) {
       output_format_ = OutputFormat::OPENTELEMETRY;
-    } else if (output_format_string == JSON_FORMAT_STR) {
-      logger_->log_trace("OutputFormat is configured to be JSON");
+      pretty_output_ = output_format_string == PRETTY_OPEN_TELEMETRY_FORMAT_STR;
+    } else if (output_format_string == PRETTY_JSON_FORMAT_STR || output_format_string == COMPACT_JSON_FORMAT_STR) {
       output_format_ = OutputFormat::JSON;
+      pretty_output_ = output_format_string == PRETTY_JSON_FORMAT_STR;
     } else {
-      logger_->log_error("Invalid OutputFormat, defaulting to JSON");
-      output_format_ = OutputFormat::JSON;
+      throw std::invalid_argument(("Invalid PerformanceDataMonitor Output Format: %s", output_format_string));
     }
   }
+  logger_->log_trace("OutputFormat is configured to be %s %s", pretty_output_ ? "pretty" : "compact", output_format_ == OutputFormat::JSON ? "JSON" : "OpenTelemtry");
+}
+
+void PerformanceDataMonitor::setupDecimalPlacesFromProperties(const std::shared_ptr<core::ProcessContext>& context) {
+  std::string decimal_places_str;
+  if (context->getProperty(DecimalPlaces.getName(), decimal_places_str)) {  // TODO(mzink): Implement int8_t in Value.h to convert int8_t implicitly
+    try {
+      decimal_places_ = std::stoi(decimal_places_str);
```

Comment:
watch out as `std::stoi` parses "dubious" strings as well (like `"-1 banana"`), using the `ValueParser` utility could elide this (unintented?) behavior

Reply:
ValueParser unfortunetly doesnt (yet) support int8_t only uint64_t, int64_t, uint32_t.
We could ofcourse use int64_t, then cast that into int8_t. If thats better?

Reply:
yes, we could go for that, and throw on out-of-range as well

Reply:
changed it in [87a9d18](https://github.com/martinzink/nifi-minifi-cpp/commit/87a9d18114347303156fd987ac482949fba47630)